src/pom.xml

```diff
@@ -1581,7 +1581,7 @@
           <printSummary>true</printSummary>
           <testFailureIgnore>${allow.test.failure.ignore}</testFailureIgnore>
           <excludedGroups>${test.excludedGroups}</excludedGroups>
-          <argLine>@{argLine} -XX:+IgnoreUnrecognizedVMOptions --illegal-access=debug -Dfile.encoding=UTF-8</argLine>
+          <argLine>@{argLine} -XX:+IgnoreUnrecognizedVMOptions -Dfile.encoding=UTF-8 --add-opens=java.base/java.lang=ALL-UNNAMED</argLine>
```

Comment:
I'm guessing that without this one XStream would not work (among other things)?

Reply:
@aaime jury is still out on that. I have no idea what xstream is used for, do you know what absolutely would break if it didn't workas user facing GeoServer functionality? Because I have run GeoServer on this branch and the java17 geotools branch locally and things seems to work to me (with no add opens).

Reply:
Well, saving some configurations would break, but I don't know, off the top of my head, which one.

Reply:
I tried doing a bunch of configuration changes and stuff and it seemed to work.